sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -261,7 +261,7 @@ final class Decimal extends Ordered[Decimal] with Serializable {
 
   def - (that: Decimal): Decimal = Decimal(toBigDecimal - that.toBigDecimal)
 
-  def * (that: Decimal): Decimal = Decimal(toBigDecimal * that.toBigDecimal)
+  def * (that: Decimal): Decimal = Decimal(toJavaBigDecimal.multiply(that.toJavaBigDecimal))
```

Comment:
It can't. But `85070591730234615847396907784232501249` can be handled by precision 38. As the `MathContext` has only precision 34, we will have scale -4 and get different result.